addons/project/models/project.py

```diff
@@ -943,6 +963,17 @@ def _notify_email_header_dict(self):
         return headers
 
     def _message_post_after_hook(self, message, msg_vals):
+        if message.attachment_ids:
+            image_count = self.env['ir.attachment'].search_count([
+                ('id', 'not in', message.attachment_ids.ids),
+                ('res_model', '=', 'project.task'),
+                ('res_id', '=', self.id),
+                ('mimetype', 'ilike', 'image')
+            ])
+            if not image_count:
+                attached_image = message.attachment_ids.search([('mimetype', 'ilike', 'image')], limit=1)
```

Comment:
search is an api.model method, so here you search an image, whatever the linked document right ?

Reply:
+ you make 2 calls to search on the attachments.

Reply:
Suggested:
```python3
    def _message_post_after_hook(self, message, msg_vals):
        if message.attachment_ids and not self.displayed_image_id:
            image_attachments = message.attachment_ids.filtered(lambda a: a.mimetype == 'image')
            if image_attachments:
                self.displayed_image_id = image_attachments[0]
```